### PR TITLE
[BugFix][Worker] Align MRV2 dummy LM-head TP before EPLB

### LINKED ISSUE #17232: [Bug] MRV2 LM-head TP and EPLB collective order hangs idle DP ranks
### Description

The QWEN3_235B_PD nightly job hangs on its first AISBench warmup request when Model Runner V2, LM-head tensor parallelism, and EPLB run together across external DP ranks. The decode API returns HTTP 500 after an HCCL timeout, the proxy returns 502, and AISBench reports `warmup failed`.

Failing run: https://github.com/vllm-project/vllm-ascend/actions/runs/35253856165/job/105314153314

### Reproduction conditions

Use vLLM Ascend commit `c7ca0b676b9668535f467b78cff1274a4ddb63b2` with vLLM 0.28.0 and the [QWEN3_235B_PD.yaml configuration at that exact commit](https://github.com/vllm-project/vllm-ascend/blob/c7ca0b676b9668535f467b78cff1274a4ddb63b2/tests/e2e/nightly/multi_node/external_dp/config/QWEN3_235B_PD.yaml). The YAML on current `main` has since changed, so the pinned revision matters.

- Hardware/topology: three 16-NPU Ascend A3 nodes; one prefill node (TP=16, DP=1) and two decode nodes (TP=1, external DP=32, 16 local DP ranks on each node, EP enabled).
- Model: `vllm-ascend/Qwen3-235B-A22B-W8A8` with `--quantization ascend`.
- Decode environment: `VLLM_USE_V1=1`, `VLLM_USE_V2_MODEL_RUNNER=1`, `HCCL_EXEC_TIMEOUT=204`, `HCCL_BUFFSIZE=1024`.
- Decode flags: `--async-scheduling --enable-expert-parallel --enable-eplb --max-model-len 36864 --max-num-batched-tokens 256 --max-num-seqs 28`.
- Decode EPLB: `--eplb-config '{"window_size":4000,"step_interval":200,"num_redundant_experts":0,"use_async":true,"log_balancedness":true,"log_balancedness_interval":1}'`.
- Ascend options: `--additional-config '{"finegrained_tp_config":{"lmhead_tensor_parallel_size":8},"eplb_config":{"load_collection_phase":"all"},"enable_mlapo":true}'`.
- Decode graph: `--compilation-config '{"cudagraph_mode":"FULL_DECODE_ONLY","cudagraph_capture_sizes":[6,12,18]}'`.
- PD transport: `MooncakeConnectorV1`, prefill `kv_producer` (DP=1, TP=16), decode `kv_consumer` (DP=32, TP=1), `use_ascend_direct=true`; the pinned YAML contains the complete connector settings.
- Trigger: run the YAML's AISBench GSM8K chat accuracy case (`batch_size=64`, `temperature=0.6`, `top_p=0.95`, `max_out_len=32768`). It fails during the first warmup request, while one decode DP rank handles work and other ranks run dummy steps. This is the observed reproducer; the individual flags have not been minimized experimentally.

### Observed behavior and diagnosis

The decode logs show DP0 entering an eight-rank LM-head `AllGather` at 17:55:37, while idle DP ranks entered a 32-rank EPLB `AllReduce` starting at 17:55:31. HCCL reports a 204-second collective timeout at 17:59:03. The first warmup request then fails with proxy `Bad Gateway`.

On the active rank, `sample()` calls `compute_logits()` (LM-head collective) before its EPLB step. On idle ranks, the V2 `_dummy_run()` override calls the parent first; the parent advances EPLB before the override calls `compute_logits()`. The two rank paths therefore enter their LM-head and EPLB collectives in opposite order. This ordering explains the timeout and is consistent with the per-rank HCCL traces.

### Expected behavior

Idle and active decode ranks should join LM-head TP before advancing the EPLB step, so the warmup request completes without an HCCL timeout.

### PR DESCRIPTION
**Pull request by freyfwt**
### What this PR does / why we need it?

Fixes #17232.

When a decode DP rank is idle, the MRV2 dummy path currently enters the parent `_dummy_run()` EPLB step before it joins the LM-head TP collective. An active rank samples through LM-head TP first and advances EPLB afterward. In the QWEN3_235B_PD nightly configuration, this opposite collective order produces an HCCL timeout.

Keep the existing dummy `compute_logits()` call after the parent's dummy forward and speculator. Pass `skip_eplb=True` to the parent's decorated `_dummy_run()`, then apply the same upstream `step_eplb_after(is_dummy=True)` decorator to the Ascend override. The resulting order is forward, dummy LM-head, EPLB, without unwrapping or changing upstream methods. Preserve `skip_eplb`, profiling, non-last-stage, and fixed logits capacity behavior. Add regression tests for the collective order and skip paths.

Restore MRV2 coverage in both Qwen3-235B PD nightly configurations associated with the failing jobs. Their decode nodes set `VLLM_USE_V2_MODEL_RUNNER=1` and enable EPLB through `--enable-eplb` and the upstream `--eplb-config` schema. Set `step_interval=2000`; omit balancedness logging options because logging is off by default. Remove Ascend's `additional_config.eplb_config` because its load collection phase defaults to `all`, along with the V1-only `DYNAMIC_EPLB` and `EXPERT_MAP_RECORD` settings. Prefill keeps its existing runner; `VLLM_USE_V1=1` selects the engine and remains shared by both roles.

The observed reproduction, including the pinned YAML and required PD, DP, EP, EPLB, and LM-head TP settings, is in #17232. This change overlaps the target-side dummy LM-head ordering part of #15079; that PR also covers speculative draft alignment.

### Does this PR introduce _any_ user-facing change?

Yes. The affected MRV2 PD deployment can complete requests without this collective-order timeout.

### How was this patch tested?

- `PYTHONPATH=<this checkout>:<vLLM checkout at 84030bbe3d74d99bad477a3d2e37a973ccd8865c> python -m pytest -q -p no:cacheprovider tests/ut/worker/test_model_runner_v2_finegrained_tp.py tests/ut/worker/test_adaptive_verification.py` — 33 passed.
- `bash format.sh ci` — passed.
- Full mypy for Python 3.10, 3.11, and 3.12 in the project lint environment — passed.
- Parsed both nightly YAML files and checked all four decode templates for MRV2, the 2000-step EPLB interval, omitted logging and Ascend EPLB options, and LM-head TP; prefill configuration remains intact.

The exact three-node NPU nightly deployment has not been rerun with this patch; the linked failing job provides the original hardware trace.

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

**Comment by yjyang62** [comment]
/nightly QWEN3_235B_PD QWEN3_235B_PD_3_5K_1_5k
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35721654886)

**Review by yiz-liu — APPROVED** [review]
